sgkit/tests/test_popgen.py

```diff
@@ -456,3 +458,132 @@ def test_Garud_h__raise_on_no_windows():
+                ],
+                [
+                    [0, 0, 0, 0],
+                    [0, 0, 0, 0],
+                    [0, 1, 2, 3],
+                    [0, 1, 2, 3],
+                    [0, 1, 2, 3],
+                    [4, 5, 6, 7],
+                ],
+                [
+                    [0, 0, -1, -1],
+                    [0, 1, -1, -1],
+                    [0, 0, 1, 1],
+                    [-1, -1, -1, -1],
+                    [0, -1, -1, -1],
+                    [-1, -1, -1, -1],
+                ],
+            ]
+        ).rechunk(chunks),
+    )
+    ds.call_genotype_mask.values = ds.call_genotype < 0
+    ds["sample_cohort"] = (
+        ["samples"],
+        da.asarray([0, 0, 1, 1, 2, 2]).rechunk(chunks[1]),
+    )
+    ho = observed_heterozygosity(ds)["stat_observed_heterozygosity"]
+    np.testing.assert_almost_equal(
+        ho,
+        np.array(
+            [
+                [0, 0, 0],
+                [1 / 4, 2 / 3, 2 / 3],
+                [0, 1, 1],
+                [1 / 2, 4 / 6, np.nan],
+            ]
+        ),
+    )
+
+
+@pytest.mark.parametrize("chunks", [((4,), (6,), (4,)), ((2, 2), (3, 3), (2, 2))])
+def test_observed_heterozygosity__windowed(chunks):
+    ds = simulate_genotype_call_dataset(
+        n_variant=4,
+        n_sample=6,
+        n_ploidy=4,
+    )
+    ds["call_genotype"] = (
+        ["variants", "samples", "ploidy"],
+        da.asarray(
+            [
+                [
+                    [0, 0, 0, 0],
+                    [0, 0, 0, 0],
+                    [1, 1, 1, 1],
+                    [1, 1, 1, 1],
+                    [0, 0, 0, 0],
+                    [1, 1, 1, 1],
+                ],
+                [
+                    [0, 0, 0, 0],
+                    [0, 0, 0, 1],
+                    [0, 0, 1, 1],
+                    [0, 0, 1, 1],
+                    [1, 0, 1, 0],
+                    [0, 1, 0, 1],
+                ],
+                [
+                    [0, 0, 0, 0],
+                    [0, 0, 0, 0],
+                    [0, 1, 2, 3],
+                    [0, 1, 2, 3],
+                    [0, 1, 2, 3],
+                    [4, 5, 6, 7],
+                ],
+                [
+                    [0, 0, -1, -1],
+                    [0, 1, -1, -1],
+                    [0, 0, 1, 1],
+                    [-1, -1, -1, -1],
+                    [0, -1, -1, -1],
+                    [-1, -1, -1, -1],
+                ],
+            ]
+        ).rechunk(chunks),
+    )
+    ds.call_genotype_mask.values = ds.call_genotype < 0
+    ds["sample_cohort"] = (
+        ["samples"],
+        da.asarray([0, 0, 1, 1, 2, 2]).rechunk(chunks[1]),
+    )
+    ds = window(ds, size=2)
+    ho = observed_heterozygosity(ds)["stat_observed_heterozygosity"]
+    np.testing.assert_almost_equal(
+        ho,
+        np.array(
+            [
+                [1 / 4, 2 / 3, 2 / 3],
+                [1 / 2, 5 / 3, np.nan],
+            ]
+        ),
+    )
```

Comment:
It would be nice to add a test to check that we're computing the same value as scikit allele for a few simulated examples. Can we compare to https://scikit-allel.readthedocs.io/en/v0.9.1/stats.html#allel.stats.heterozygosity_observed ?